components/molecules/FilterCardSelect/filter-card-select.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useRef, useState } from "react";
+import Image from "next/image";
+import Text from "../../atoms/Typography/text";
+import hashIcon from "../../../img/icons/hash.svg";
+import orgIcon from "../../../img/icons/org.svg";
+import personIcon from "../../../img/icons/person.svg";
+import repoIcon from "../../../img/icons/repo.svg";
+import Selector from "../../atoms/Selector/selector";
+
+interface FilterCardSelectProps {
+    selected: string;
+    icon?: "topic" | "repo" | "org" | "contributor";
+    options: string[];
+    handleFilterClick: (filter: string) => void;
+}
+
+const icons = {
+  topic: {
+    src: hashIcon.src,
+    alt: "Topic"
+  },
+  org: {
+    src: orgIcon.src,
+    alt: "Organization"
+  },
+  contributor: {
+    src: personIcon.src,
+    alt: "Contributor"
+  },
+  repo: {
+    src: repoIcon.src,
+    alt: "Repository"
+  }
+};
+
+const FilterCardSelect: React.FC<FilterCardSelectProps> = ({ selected: filterName, icon = "topic", options, handleFilterClick }) => {
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleFilter = () => {
+    setIsOpen(!isOpen);
+  };
+
+  useEffect(() => {
+    const checkIfClickedOutside = (e: any) => {
```

Comment:
This should not be `e: any`. 

```suggestion
    const checkIfClickedOutside = (e: MouseEvent) => {
```